### ISSUE TITLE
Unclear instructions on how to upload test evidence

### ISSUE THREAD

**Issue by RicardoM17** [fundakol/pytest-jira-xray#88]
Hello @fundakol , thanks a lot for this tool. It seems like it will save me a lot of hassle.

I'm having some issues understanding how to properly attach evidence to a test report which perhaps you can help me with.

My use case is to have the tests run in a bitbucket pipeline which will then upload the results to a new XRAY execution.

I already have that bit working and the new execution is created for each run, I could even correctly set the version.

However I am having some issues with adding evidence to these test runs.

My test_script so far is very much like the one you have in your example:

```import pytest

@pytest.mark.xray('JIRA-1')
def test_foo():
    assert True```

However following your example in the [documentation](https://github.com/fundakol/pytest-jira-xray#attach-test-evidences) I can't really get it to work. I'm also struggling a bit to understand how this hook is supposed to work, when is it supposed to be called, and how can I add evidences from the tests. Where is the `outcome` supposed to come from, how can each test add to the evidences etc etc. Right now I'm mostly just focusing on the simple stuff like the log output, so nothing fancy.

In the pipeline I run the following commands:

```
pytest --junitxml=test-result.xml dummy_test.py
pytest --jira-xray --client-secret-auth --cloud
```

Also I would like to know what is the recommended way to set the env variables like `XRAY_EXECUTION_SUMMARY` or if there's an alternative and convenient way to do this from within the test script for example.

Many thanks in advance!!

**Comment by fundakol** [comment]
If you want to know how hooks work you should check pytest documentation (e.g. https://docs.pytest.org/en/7.1.x/how-to/writing_hook_functions.html)

But shortly, this `pytest_runtest_makereport` hook is called for each test (for setup, run/call and teardown). So if a test, for example failed you can check its status and attached some evidence e.g. some log files or screenshot if it was web application.

**Comment by RicardoM17** [comment]
Hi @fundakol thank you for your quick reply.

I searched a bit more, including with your link and found the issue. I didn't have `pytest_runtest_makereport` in `conftest.py` file. Instead I had it in the same file as in the tests.

This is in the documentation you have as there is `# FILE: conftest.py` right at the top but I still missed it :sweat_smile: 

However I wasn't the only one, as I found another soul that made the same mistake as me. Hiding in plain sight.

It's working now!

Thanks once again for all the help.
